pkg/controllers/cloud/cloudentityselector_controller.go

```diff
@@ -33,7 +35,9 @@ import (
 )
 
 const (
-	virtualMachineIndexerByCloudAccount = "virtualmachine.cloudaccount"
+	virtualMachineIndexerByCloudAccount     = "virtualmachine.cloudaccount"
+	virtualMachineSelectorMatchIndexerByVPC = "virtualmachine.selector.vpc"
+	virtualMachineSelectorMatchIndexerByID  = "virtualmachine.selector.id"
```

Comment:
We would need indexer for VPC name and VM Name. I believe uniqueness of the indexer will be problem. Probably we can add webhook validation for the same

Reply:
Added the indexer for VM and VPC names. 
Yes, we can add webhook validations.

Reply:
Based on offline discussion, decided to add it in a later commit.